### ISSUE TITLE
Formatter undocumented deviation: condition and slice wrapping

### ISSUE THREAD

**Issue by JonathanPlasse** [astral-sh/ruff#7319]
Black formatting
```python
def f():
    return (
        package_version is not None
        and package_version.split(".")[:2] == package_info.version.split(".")[:2]
    )
```
Ruff formatting
```python
def f():
    return package_version is not None and package_version.split(".")[
        :2
    ] == package_info.version.split(".")[:2]
```
I use Ruff 0.0.289 with line length 100.

In this case, I prefer Black.

**Comment by MichaReiser** [comment]
Thanks for reporting this deviation. I'm, unfortunately, unable to reproduce the deviation. Both Black and Ruff create the same formatted output for me. 

```
❯ pipx run ruff format test.py
warning: `ruff format` is a work-in-progress, subject to change at any time, and intended only for experimentation.
1 file left unchanged

~/astral/test via 🐍 v3.11.5 
❯ bat test.py
───────┬──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       │ File: test.py
───────┼──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ def f():
   2   │     return (
   3   │         package_version is not None
   4   │         and package_version.split(".")[:2] == package_info.version.split(".")[:2]
   5   │     )
───────┴──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

```

**Comment by JonathanPlasse** [comment]
```console
❯ cat haha.py
def f():
    return (
        package_version is not None
        and package_version.split(".")[:2] == package_info.version.split(".")[:2]
    )
❯ pipx run ruff format haha.py --isolated
warning: `ruff format` is a work-in-progress, subject to change at any time, and intended only for experimentation.
1 file reformatted
❯ cat haha.py
def f():
    return package_version is not None and package_version.split(".")[
        :2
    ] == package_info.version.split(".")[:2]
```

**Comment by MichaReiser** [comment]
[Playground](https://play.ruff.rs/7a0ebba3-4261-4ca6-ad4c-e2d9a3088ea9)
